python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion_with_data.py

```diff
@@ -0,0 +1,303 @@
+from semantic_kernel.connectors.ai.open_ai.services.open_ai_model_types import OpenAIModelTypes
+from semantic_kernel.sk_pydantic import HttpsUrl
+
+class AzureChatCompletionWithData(AzureChatCompletion):
+    _data_source_settings: AzureChatWithDataSettings
+
+    def __init__(
+        self,
+        deployment_name: str,
+        endpoint: Optional[Union[HttpsUrl, str]] = None,
+        api_key: Optional[str] = None,
+        api_version: str = DEFAULT_AZURE_WITH_DATA_API_VERSION,
+        data_source_settings: Optional[AzureChatWithDataSettings] = None,
+        ad_token: Optional[str] = None,
+        ad_token_provider: Optional[AsyncAzureADTokenProvider] = None,
+        log: Optional[Logger] = None,
+    ) -> None:
+        """
+        Initialize an AzureChatCompletionWithData service.
+
+        You must provide:
+        - A deployment_name, endpoint, api_key, and datasource_settings (plus, optionally: ad_token or ad_token_provider)
+
+        :param deployment_name: The name of the Azure deployment. This value
+            will correspond to the custom name you chose for your deployment
+            when you deployed a model. This value can be found under
+            Resource Management > Deployments in the Azure portal or, alternatively,
+            under Management > Deployments in Azure OpenAI Studio.
+        :param endpoint: The endpoint of the Azure deployment. This value
+            can be found in the Keys & Endpoint section when examining
+            your resource from the Azure portal.
+        :param api_key: The API key for the Azure deployment. This value can be
+            found in the Keys & Endpoint section when examining your resource in
+            the Azure portal. You can use either KEY1 or KEY2.
+        :param api_version: The API version to use. (Optional)
+            The default value is "2022-12-01".
+        :param data_source_settings: The settings for the data source.
+        :param log: The logger instance to use. (Optional)
+        :param ad_token: The Azure AD token to use. (Optional)
+        :param ad_token_provider: The Azure AD token provider to use. (Optional)
+        """
+        if not deployment_name:
+            raise ValueError("The deployment name cannot be `None` or empty")
+        
+        base_url = f"{str(endpoint).rstrip('/')}/openai/deployments/{deployment_name}/extensions"
+
+        if isinstance(endpoint, str):
+            endpoint = HttpsUrl(endpoint)
+        if not data_source_settings:
+            raise ValueError("The data source settings cannot be `None`")
+
+        super().__init__(
+            deployment_name,
+            endpoint=endpoint,
+            base_url=base_url,
+            api_version=api_version,
+            api_key=api_key,
+            log=log,
+            ad_token=ad_token,
+            ad_token_provider=ad_token_provider,
+        )
+
+        self._data_source_settings = data_source_settings
+
+    async def complete_chat_async(
```

Comment:
Is there any way we can re-use the code that exists in `open_ai_chat_completion_base.py`? Most of the code shown here like `complete_chat_async`, `complete_chat_stream_async`, `_parse_choices` and others exist there.